### PR TITLE
Added Metadata service, added a way to obtain View Billing Portal Management Link

### PR DESCRIPTION
**Pull request by teodortalov**
- Added Metadata service that populates data into pre-defined in Chargify UI metafields. Currently, only support for populating those fields is added. 
- Added a way to obtain View Billing Portal Management Link for a given customer. View Billing Portal Management Link are the hosted pages where customer can manage their account.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,40 +1,38 @@
 {
```
Comment (dan-bowen) [inline composer.json:1]: Please revert changes to this file.

File: src/Crucial/Service/Chargify.php [hunk 2]
```diff
@@ -254,6 +255,15 @@ public function charge()
     }
 
     /**
+     * Helper for instantiating an instance of Metadata
+     *
+     * @return Charge
```
Comment (dan-bowen) [inline src/Crucial/Service/Chargify.php:260]: Return type should be `Metadata`.

File: src/Crucial/Service/Chargify/Metadata.php [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+class Metadata extends AbstractEntity
+{
+
+    /**
+     * For subscriptions you have the ability to add / populate metadata
+     * (custom fields) into pre-defined metafields.
+     *
+     * Pass data to custom fields using the AbstractEntity::setParam() method
+     * @param int $subscriptionId
```
Comment (dan-bowen) [inline src/Crucial/Service/Chargify/Metadata.php:13]: Please add a newline before the `@param`.

**Comment by dan-bowen** [comment]
Hello. Thank you for your interest in this project. I have made a few comments that would need to be addressed in order for this to be accepted. Also, could you add some tests around this new functionality? Please refer to existing test here. https://github.com/chargely/chargify-sdk-php/tree/master/tests/phpunit

**Comment by dan-bowen** [comment]
Also, this PR will need to target the `develop` branch, not `master`. Thanks.

**Comment by teodortalov** [comment]
@crucialwebstudio , thanks for your feedback. I will implement the changes you recommended and add some tests and update the PR to develop.

Thanks :)

**Pull request closed without merging**